warehouse/oidc/tasks.py

```diff
@@ -0,0 +1,24 @@
+from warehouse import tasks
+from warehouse.metrics import IMetricsService
+from warehouse.packaging.models import Project
+
+
+@tasks.task(ignore_result=True, acks_late=True)
+def compute_oidc_metrics(request):
+    metrics = request.find_service(IMetricsService, context=None)
+
+    projects_using_oidc = (
+        request.db.query(Project).distinct().join(Project.oidc_publishers)
+    )
+
+    # Metric for count of all projects using OIDC
+    metrics.gauge(
+        "warehouse.oidc.total_projects_using_oidc_publishers",
+        projects_using_oidc.count(),
```

Comment:
Also noticed the usage of `.count()` - which is slower than postgres' `count` function (I know, it's confusing...)

Since we don't intend to use the returned objects in any way, we can use `func.count()` to add the `COUNT(*)` to the SQL query. See https://docs.sqlalchemy.org/en/20/core/functions.html#sqlalchemy.sql.functions.count

Reply:
This is not a hard requirement - we use count() elsewhere already with no material negative impact observed thus far. I saw something, so I said something. 😉